### DISCUSSION TITLE
Bad Request: failed to process audio: corrupt or unsupported data

### DISCUSSION THREAD

**Discussion by midou221** [deepgram/community#181] [General help] [answered]
`{"err_code":"Bad Request","err_msg":"Bad Request: failed to process audio: corrupt or unsupported data","request_id":"514f0042-b572-48c2-833c-4be5eb6b77c9"}`
I always get this response when I try to send a request..
I'm using the Okhttp library and send the mp3 file through it.. 
```
val pickAudio = registerForActivityResult(ActivityResultContracts.GetContent()) { uri ->
        val file = getFileFromUri(uri!!)
        if(file!!.exists()) {
            NetworkHelper.sendCall(file).enqueue(object : Callback {
                override fun onFailure(call: Call, e: IOException) {
                    e.printStackTrace()
                    Log.e(TAG, "onFailure ${e.message}")

                }

                override fun onResponse(call: Call, response: Response) {
                    Log.e(TAG, "onResponse ${response.body.string()}")

                }

            })
        }

    }

   fun sendCall(audioFile: File): Call {
        val requestBody: RequestBody = MultipartBody.Builder()
            .setType(MultipartBody.FORM)
            .addFormDataPart(
                "file", audioFile.name,
                audioFile.asRequestBody("audio/mp3".toMediaTypeOrNull())
            )
            .build()

        val request: Request = Request.Builder()
            .url("https://api.deepgram.com/v1/listen")
            .header("Authorization", "Token ****************************")
            .post(requestBody)
            .build()

        return client.newCall(request)
    }

```

 honestly I can't figure out the bug..
I hope there will be support for Kotlin or Java like other languages

**Comment by jjmaldonis** [comment, marked as answer]
Hey @midou221! A `Bad request` error message is often due to some part of the request being formatted incorrectly.

One thing you can change right away is using `audio/mpeg` rather than `audio/mp3` because technically `audio/mp3` isn't a mime type.

Also, when an audio file is loaded into a byte array and passed to the Deepgram API, the key:value pair that goes in the `data` part of the request is: `{"source": <your audio byte array>}`. So if you change `"file"` to `"source"` that will help. You'll also want to include the mime type in the `data` part of the request, so the full `data` part will look like this:

```
{"source": <your audio byte array>, "mimetype": "audio/mpeg"}
```

It's also possible that the `Bad request` message could be coming from the URI, so if neither of the above are fixing the issue, could you copy-paste the URI so we can take a look?
